arangod/IResearch/IResearchLink.cpp

```diff
@@ -723,95 +724,94 @@ arangodb::Result IResearchLink::drop() {
-    );
+      "caught exception while removing arangosearch link '" + std::to_string(id()) + "': " + e.what()
+    };
   } catch (...) {
-    return arangodb::Result(
+    return {
       TRI_ERROR_INTERNAL,
-      std::string("caught exception while removing arangosearch link '") + std::to_string(id()) + "'"
-    );
+      "caught exception while removing arangosearch link '" + std::to_string(id()) + "'"
+    };
   }
 
-  return arangodb::Result();
+  return {};
 }
 
 bool IResearchLink::hasSelectivityEstimate() const {
   return false; // selectivity can only be determined per query since multiple fields are indexed
 }
 
-arangodb::Result IResearchLink::init(
-    arangodb::velocypack::Slice const& definition,
+Result IResearchLink::init(
+    velocypack::Slice const& definition,
     InitCallback const& initCallback /* = { }*/ ) {
   // disassociate from view if it has not been done yet
   if (!unload().ok()) {
-    return arangodb::Result(TRI_ERROR_INTERNAL, "failed to unload link");
+    return { TRI_ERROR_INTERNAL, "failed to unload link" };
   }
 
   std::string error;
   IResearchLinkMeta meta;
 
-  if (!meta.init(definition, true, error, &(collection().vocbase()))) { // definition should already be normalized and analyzers created if required
-    return arangodb::Result(
+  // definition should already be normalized and analyzers created if required
+  if (!meta.init(definition, true, error, &(collection().vocbase()))) {
+    return {
       TRI_ERROR_BAD_PARAMETER,
-      std::string("error parsing view link parameters from json: ") + error
-    );
+      "error parsing view link parameters from json: " + error
+    };
   }
 
   if (!definition.isObject() // not object
       || !definition.get(StaticStrings::ViewIdField).isString()) {
-    return arangodb::Result( // result
-      TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND, // code
-      std::string("error finding view for link '") + std::to_string(_id) + "'" // message
-    );
+    return {
+      TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND,
+      "error finding view for link '" + std::to_string(_id) + "'"
+    };
   }
 
   auto viewId = definition.get(StaticStrings::ViewIdField).copyString();
   auto& vocbase = _collection.vocbase();
   bool const sorted = !meta._sort.empty();
 
-  if (arangodb::ServerState::instance()->isCoordinator()) { // coordinator link
-    auto* ci = arangodb::ClusterInfo::instance();
+  if (ServerState::instance()->isCoordinator()) { // coordinator link
+    auto* ci = ClusterInfo::instance();
 
     if (!ci) {
-      return arangodb::Result( // result
-        TRI_ERROR_INTERNAL, // code
-        std::string("failure to get storage engine while initializing arangosearch link '") + std::to_string(_id) + "'"
-      );
+      return {
+        TRI_ERROR_INTERNAL,
+        "failure to get storage engine while initializing arangosearch link '" + std::to_string(_id) + "'"
```

Comment:
Error message is misleading, as it is actually not the storage engine that is missing but the cluster info.

Reply:
done